### DISCUSSION TITLE
[@types/jsonwebtoken]: types not reflect cjs export

### DISCUSSION THREAD

**Discussion by trim21** [DefinitelyTyped/DefinitelyTyped#65458] [Issues with a @types package]
https://github.com/auth0/node-jsonwebtoken/blob/master/index.js

```ts
module.exports = {
  decode: require('./decode'),
  verify: require('./verify'),
  sign: require('./sign'),
  JsonWebTokenError: require('./lib/JsonWebTokenError'),
  NotBeforeError: require('./lib/NotBeforeError'),
  TokenExpiredError: require('./lib/TokenExpiredError'),
};
```

https://github.com/trim21/DefinitelyTyped/blob/53a5a31dcf843ccde716b4a1a498f7b0fc6de62e/types/jsonwebtoken/index.d.ts#L251-L253

we should use a `export =` here

```ts
interface JWTStatic {
    decode: typeof decode;
    verify: typeof verify;
    sign: typeof sign;
    JsonWebTokenError: typeof JsonWebTokenError;
    NotBeforeError: typeof NotBeforeError;
    TokenExpiredError: typeof TokenExpiredError;
}

declare const jwtStatic: JWTStatic;

export = jwtStatic;
```

But there are also some types packages import types from this package `jsonwebtoken`
```ts
import { SignOptions } from 'jsonwebtoken';
```

So looks like we don't have a correct way to write types for this package? 

I try to export namespace, but it's skill valid to `import * as jwt from 'jsonwebtoken';jwt.sign(...)`

looks like this can be fixed by https://github.com/auth0/node-jsonwebtoken/pull/884

[a comment by an automated account is not shown]

**Comment by peterblazejewicz** [comment]
@trim21 yes, there is no 1-to-1 pattern matched here, but exports are matching module exports, as the results there is no real difference when consuming the types. What's the problem you have with this DT definition?

**Reply by trim21** [reply]
I have a problme but looks like it doesn't have a solution, because there is no way to export types with export assignment syntax.

ts code with tsconfig `"esModuleInterop": false,` and `"allowSyntheticDefaultImports": false,` should throw type error (it's also not valid in (at least) nodejs runtime)

```ts
import * as jwt from 'jsonwebtoken'
jwt.sign(...)
```
